ServerManager/classes/ServerManager.php

```diff
@@ -291,9 +291,9 @@ public function getServerURLBySessionId($sessionId = '', bool $forDocker = false
     {
         // e.g. http://localhost/1
         // use this one if you just want the full URL of a Server's session
-        $url = $forDocker ?
-            // this is always called from inside the docker environment,so just use http://caddy:80/...
-            'http://caddy:80'.Config::get('code_branch') :
+        $url = $forDocker ? // in-case of docker, use http://caddy:80/ or http://mitmproxy:8080/
+            'http://'.($_ENV['WEB_SERVER_HOST'] ?? 'caddy').':'.($_ENV['WEB_SERVER_PORT'] ?? 80).
+            Config::get('code_branch') :
```

Comment:
please sift this out for a separate merge into dev so it can become part of 4.0.2